src/project-manager.ts

```diff
@@ -761,14 +761,18 @@ export class InMemoryFileSystem implements ts.ParseConfigHost, ts.ModuleResoluti
 	 * @param path file path (both absolute or relative file paths are accepted)
 	 */
 	fileExists(path: string): boolean {
-		return this.readFile(path) !== undefined;
+		return this.readFileImpl(path) !== undefined;
 	}
 
 	/**
 	 * @param path file path (both absolute or relative file paths are accepted)
-	 * @return file's content in the following order (overlay then cache) if any
+	 * @return file's content in the following order (overlay then cache)
 	 */
-	readFile(path: string): string | undefined {
+	readFile(path: string): string {
+		return this.readFileImpl(path) || '';
```

Comment:
This looks like something that could cause hard to track bugs. Would it work to throw an error instead?